### PR TITLE
Exclude Crawler from url usage count

### PR DESCRIPTION
**Pull request by sal-himd**

### CODE REVIEW

**Comment by jpmcgrath** [comment]
@sal-himd looks great!

One thing: Can I ask where the list of user agent strings came from? I maintaining a list of bot strings inside this gem could be a bit of a pain. I think it would be good to outsource this to a gem specialising in bot detection. Both of these seem like well maintained Gems:
https://rubygems.org/gems/voight_kampff
https://rubygems.org/gems/browser

Thoughts?

**Comment by sal-himd** [comment]
I totally agree with you. I will do it as soon as I have the time.

**Comment by sal-himd** [comment]
Done.

Tell me if you have anything else to say about my PR. I can rebase my commits history if you want to merge it.

**Comment by jpmcgrath** [comment]
This looks really good. Thanks. I'm thinking that to make this perfect, it would be ideal to be able to turn this on/off in the initializer (and document it in the readme).

The one thing that I am torn on is whether it should be on by default, or off. I think that we should default it to on, and bump the versioning to indicate a change in behaviour, along with appropriate mention in the readme of the change.

What do you think?

**Comment by sal-himd** [comment]
I think you right. Switchable and on by default would be perfect for me, but may be you should consult other users of your gem ?

**Comment by jpmcgrath** [comment]
@sal-himd I'd like to get this merged in, and have it turned off by default. That way people can turn it on if they want it on. Then down the line we can look at making it the default behaviour after a bit more discussion. 

Do you think you could update the PR with the ability to turn on/off with the default to off?

Cheers.

**Comment by sal-himd** [comment]
I will do it this week end.

**Comment by simon02** [comment]
This is awesome, already created my own fork using voight-kampff. Will come back to this one once it is merged in.

**Comment by jpmcgrath** [comment]
@sal-himd I realised that this great edition never got merged. I've gone ahead and merged this along with providing a `ignore_robots` config attribute. Thanks for the great edition.

@simon02 you might also be interested that this feature is now released as v0.5.6 https://rubygems.org/gems/shortener/versions/0.5.6

Cheers.

File: shortener.gemspec [hunk 1]
```diff
@@ -15,6 +15,7 @@ Gem::Specification.new do |s|
   s.rubyforge_project         = "shortener"
   s.required_rubygems_version = "> 1.3.6"
   s.add_dependency "rails", ">= 3.0.7"
+  s.add_dependency "voight_kampff", '~>1.0.2'
```
Comment (zillou) [inline shortener.gemspec]: @blackbirdco @jpmcgrath is it possible to relax the version of `voight_kampff`, as voight_kampff version 1.0.x depends on rack 1.0, but rails 5 depend on rack 2, which makes shortener cannot be used in rails 5 projects.

**Pull request merged**